eng/common/scripts/Delete-RemoteBranches.ps1

```diff
@@ -66,67 +65,124 @@ if ($AuthToken) {
 }
 
 $owner, $repo = $RepoId -split "/"
-$branches = Get-AllBranchesAndPullRequestInfo $owner $repo
 
-foreach ($branch in $branches)
-{
-  $branchName = $branch.Name
-  if ($branchName -notmatch $BranchRegex) {
-    continue
+# These will always be output at the end of the script. Their only purpose is for information gathering
+# Total number returned from query
+$totalBranchesFromQuery = 0
+# reasons why a branch was skipped
+$skippedBranchNotMatchRegex = 0
+$skippedForCommitDate = 0
+$skippedForOpenPRs = 0
+$skippedForPRNotInBranch = 0
+$skippedForPRNotInRepo = 0
+# gh call counters
+$ghPRViewCalls = 0
+$ghBranchDeleteCalls = 0
+
+try {
+  # Output the core rate limit at the start of processing. There's no real need
+  # to output this at the end because the GH call counts are being output
+  $coreRateLimit = Get-RateLimit ([RateLimitTypes]::core)
```

Comment:
While this works for 1 parameter you shouldn't call PS functions with `(...)` as it turns them into an array.

Reply:
For an enum, this isn't creating an array it's evaluating it. The syntax and processing for enums in powershell is a bit weird. That being said, after a little tinkering, I found `Get-RateLimit ([RateLimitTypes]::core)` is the same as `Get-RateLimit core` and I'll change them to that since it looks less...odd.